src/test/java/net/consensys/orion/network/PersistentNetworkNodesTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.network;
+
+import static org.apache.tuweni.crypto.sodium.Box.KeyPair.random;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import net.consensys.orion.config.Config;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.tuweni.crypto.sodium.Box;
+import org.apache.tuweni.kv.MapKeyValueStore;
+import org.junit.jupiter.api.Test;
+
+class PersistentNetworkNodesTest {
```

Comment:
Could we add a test to ensure that we don't override a known node entry? This is to ensure we never allow discovery to override the table.

Reply:
sure thing, adding that now.